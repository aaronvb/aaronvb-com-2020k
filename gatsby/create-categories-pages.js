'use strict';

const _ = require('lodash');
const path = require('path');

module.exports = async (graphql, actions) => {
  const { createPage } = actions;

  const result = await graphql(`
    {
      allMarkdownRemark(
        filter: { frontmatter: { template: { eq: "post" }, draft: { ne: true } } }
      ) {
        group(field: frontmatter___category) {
          fieldValue
          totalCount
        }
      }
    }
  `);

  _.each(result.data.allMarkdownRemark.group, (category) => {
    const categorySlug = `/category/${_.kebabCase(category.fieldValue)}`;

    createPage({
      path: categorySlug,
      component: path.resolve('./src/templates/category-template.js'),
      context: {
        category: category.fieldValue,
      }
    });
  });
};
