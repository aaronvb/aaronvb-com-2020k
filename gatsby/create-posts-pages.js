'use strict';

const path = require('path');

module.exports = async (graphql, actions) => {
  const { createPage } = actions;

  createPage({
    path: '/',
    component: path.resolve('./src/templates/index-template.js'),
  });
};
