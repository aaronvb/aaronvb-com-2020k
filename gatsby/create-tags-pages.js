'use strict';

const _ = require('lodash');
const path = require('path');

module.exports = async (graphql, actions) => {
  const { createPage } = actions;

  const result = await graphql(`
    {
      allMarkdownRemark(
        filter: { frontmatter: { template: { eq: "post" }, draft: { ne: true } } }
      ) {
        group(field: frontmatter___tags) {
          fieldValue
          totalCount
        }
      }
    }
  `);

  _.each(result.data.allMarkdownRemark.group, (tag) => {
    const tagSlug = `/tag/${_.kebabCase(tag.fieldValue)}`;

    createPage({
      path: tagSlug,
      component: path.resolve('./src/templates/tag-template.js'),
      context: {
        tag: tag.fieldValue,
      }
    });
  });
};
