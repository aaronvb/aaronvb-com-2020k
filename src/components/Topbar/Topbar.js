// @flow strict
import React from 'react';
import Author from './Author';
import Menu from './Menu';
import styles from './Topbar.module.scss';
import { useSiteMetadata } from '../../hooks';

const Topbar = () => {
  const { author, menu } = useSiteMetadata();

  return (
    <div className={styles['topbar']}>
      <div className={styles['topbar__inner']}>
        <Author author={author} />
        <Menu menu={menu} />
      </div>
    </div>
  );
};

export default Topbar;
