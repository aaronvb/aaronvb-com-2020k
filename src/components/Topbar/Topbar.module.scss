@import '../../assets/scss/variables';
@import '../../assets/scss/mixins';

.topbar {
    max-width: $layout-post-width;
    margin-left: auto;
    margin-right: auto;
    &__inner {
        padding: 0 20px 0;
        @include margin-top(1);
        @include margin-bottom(1.5);
    }
}

@include breakpoint-sm {
    .topbar {
        &__inner {
            padding: 0;
            @include margin-top(2);
            @include margin-bottom(2);
            line-height: 1.8rem;
        }
    }
}

@include breakpoint-md {
    .topbar {
        &__inner {
            padding: 0;
            @include margin-top(2);
            @include margin-bottom(2);
            line-height: 1.8rem;
        }
    }
}
