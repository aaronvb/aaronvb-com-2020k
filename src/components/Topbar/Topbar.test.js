// @flow strict
import React from 'react';
import renderer from 'react-test-renderer';
import { useStaticQuery, StaticQuery } from 'gatsby';
import MockDate from 'mockdate';
import Topbar from './Topbar';
import siteMetadata from '../../../jest/__fixtures__/site-metadata';
import type { RenderCallback } from '../../types';

describe('Tobar', () => {
  beforeEach(() => {
    StaticQuery.mockImplementationOnce(
      ({ render }: RenderCallback) => (
        render(siteMetadata)
      ),
      useStaticQuery.mockReturnValue(siteMetadata)
    );
  });

  const props = {
    isIndex: true
  };

  it('renders correctly', () => {
    MockDate.set(new Date('2020-04-30T11:01:58.135Z'));
    const tree = renderer.create(<Topbar {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
    MockDate.reset();
  });
});
