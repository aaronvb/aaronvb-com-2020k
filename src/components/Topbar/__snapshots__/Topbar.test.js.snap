// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tobar renders correctly 1`] = `
<div
  className="topbar"
>
  <div
    className="topbar__inner"
  >
    <div
      className="author"
    >
      <span
        className="author__title"
      >
        <a
          className="author__title-link"
          href="/"
        >
          Test name
        </a>
      </span>
    </div>
    <nav
      className="menu"
    >
      <a
        className="menu__list-item-link"
        href="//test/1/"
      >
        Test label 1
      </a>
      <a
        className="menu__list-item-link"
        href="//test/2/"
      >
        Test label 2
      </a>
      <a
        className="menu__list-item-link"
        href="//test/3/"
      >
        Test label 3
      </a>
    </nav>
  </div>
</div>
`;
